workflow/validate/validate.go

```diff
@@ -259,6 +369,15 @@ func validateArtifactLocation(errPrefix string, art wfv1.ArtifactLocation) error
 func resolveAllVariables(scope map[string]interface{}, tmplStr string) error {
 	var unresolvedErr error
 	_, allowAllItemRefs := scope[anyItemMagicValue] // 'item.*' is a magic placeholder value set by addItemsToScope
+
+	//inputs.parameters.regexp. stores regexps for input parameters
+	inputRegexps := make([]string, 0)
+	for k := range scope {
+		if strings.HasPrefix(k, "inputs.parameters.regexp.") {
```

Comment:
This one should also be `passthroughRegexp` instead of `regexp`.

Reply:
Yes, that would be clearer. 👍